templates/injector-leader-endpoint.yaml

```diff
@@ -0,0 +1,8 @@
+{{- if and (eq (.Values.injector.enabled | toString) "true" ) (eq (.Values.global.enabled | toString) "true") (eq (.Values.injector.leaderElector.enabled | toString) "true") (gt (.Values.injector.replicas | int) 1) }}
+# This is created here so it can be cleaned up easily, since if
+# the endpoint is left around the leader won't expire for about a minute.
+apiVersion: v1
```

Comment:
This needs Helm annotations:

```yaml
     app.kubernetes.io/name: {{ include "vault.name" . }}-agent-injector
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
```

Reply:
Good shout. Should I add the same to the secret too?

Reply:
Yup, everything we make should have them.

Reply:
Done